### PR TITLE
Issue#5 Update endpoints

### LINKED ISSUE #5: Backend: think about endpoints

### PR DESCRIPTION
**Pull request by Mxx1029**
- new version of endpoints in userController.js (all of them, for users and events are in one file for simplicity)
- old endpoints commented out as version 1
- new stuff not before line 40

### CODE REVIEW

File: router/userRouter.js [hunk 1]
```diff
@@ -1,20 +1,98 @@
...
+//     `/events?category=${category}&date=${date}&=${keyword}`,
+//     getEvents(category, date, keyword)
+// );
+
+// version 2 (18/11/2021)
+import * as eventController from "../controllers/eventController.js";
+import * as userController from "../controllers/userController.js";
+router.get("/events/today", eventController.getEventsForToday); // landing page     
+router.get("/events", eventController.getEvents); // get events using req.query
+router.get("/events/:eventId", eventController.getEvent); // click on one event
```
Comment (Req) [inline router/userRouter.js:45]: Very nice! :+1:

File: router/userRouter.js [hunk 2]
```diff
@@ -1,20 +1,98 @@
...
+// router.put(`/users/${userId}`, updateUser(userId));
+// router.delete(`/users/${userId}`, deleteUser(userId));
+
+// version 2 (18/11/2021)
+router.get("/users", userController.getUsers);
+router.get("/users/:userId", userController.getUser);
+router.post("/users", userController.addUser); // register a user
+router.post("/users/:userId", userController.loginUser); // login a user --> after this, user will get events of today, where they started from 
+router.put("/users/:userId", userController.updateUser);
+router.delete("/users/:userId", userController.deleteUser);
```
Comment (Req) [inline router/userRouter.js:62]: Nice! 

Two small comments;
`router.post("/users/:userId", userController.loginUser); `
When you are doing a login, you most likely will not have the id of the user available; I would just use POST /user/login as the endpoint and the request body will contain the username/password or email/password

Also; your server.js already contains `server.use("/users", userRouter);`

So the /users is already coming from that place. So; these ones would in this case be 
GET /users/users
GET /users/users/:userId
POST /users/users
POST  /users/users/:userId
and so on... so the 2nd /users is not needed here :smile_cat:

File: router/userRouter.js [hunk 3]
```diff
@@ -1,20 +1,98 @@
...
+// version 1 (16/11/2021)
+// router.post(`/users/${userId}/events`, addEvent(userId));
+// router.put(`/users/${userId}/events/${eventId}`, updateEvent(userId, eventId));
+// router.delete(`/users/${userId}/events/${eventId}`, deleteEvent(userId, eventId));
+
+// version 2 (18/11/2021)
+
+router.post("/users/:userId/events", eventController.addEvent);
+router.put("/users/:userId/events/:eventId", eventController.updateEvent);
+router.delete("/users/:userId/events/:eventId", eventController.deleteEvent);
```
Comment (Req) [inline router/userRouter.js:96]: Overall it's looking really good! :sun_with_face:

**Comment by Mxx1029** [comment]
Going to close this too and do a new version, that is actually ready to be merged into main

**Pull request closed without merging**